### ISSUE TITLE
error after submission history activated

### ISSUE THREAD

**Issue by sklmteacher** [formtools/module-submission_accounts#5]
following error showed when I updated a record through individual login after activated submission history

Problem with this query in sh_add_history_row(): SQLSTATE[01000]: Warning: 1265 Data truncated for column 'sh___change_account_type' at row 1

**Comment by benkeen** [comment]
Hi @sklmteacher, excellent thanks for logging this. Basically the problem is that it's just attempting to write some data into that field that's too big. But this should NOT occur because that field isn't customizable. Thanks! I'll give it a look.

**Comment by sklmteacher** [comment]
I got same error even I tried to update fields with single characters each.

**Comment by benkeen** [comment]
Yeah I'd imagine it'll occur with all data changes on the form. Are you able to glance at the form table in the database? I'm looking at the field and it's an ENUM:  `enum('admin', 'client', 'unknown')` meaning it gets fed specific values. So I'm kind of puzzled by the error message.

**Comment by benkeen** [comment]
Hey @sklmteacher, I haven't been able to reproduce this I'm afraid. Let me know about the database table column structure to just confirm that's not the issue.

**Comment by sklmteacher** [comment]
I found same values enum('admin','client','unknown') NULL
at column sh___change_account_type .

**Comment by benkeen** [comment]
Cool thanks. And you're logging in as a client/admin account when you see the error?

**Comment by sklmteacher** [comment]
Through submission account

**Comment by sklmteacher** [comment]
http://kamalvilekar.com/govind.php

user: xxx
pass: yyy

**Comment by benkeen** [comment]
Ahhh! Sorry, how daft of me. Thanks.

**Comment by benkeen** [comment]
Still no luck reproducing this on my end, but I dp think I see what may be the problem. Could you try this: change the `sh___change_account_type` database field in your history table to this:

`'admin','client','user','unknown'` (note the addition of `user`). 

Then try logging out of FT, logging in via the Submission Accounts user and updating the data.

**Comment by sklmteacher** [comment]
I did, but now I got a new error,
 for database structure

http://kamalvilekar.com/adminer.php?username=ngr18

pass: ftools@18

**Comment by sklmteacher** [comment]
the error is
Problem with this query in sh_add_history_row(): SQLSTATE[HY000]: General
error: 1366 Incorrect integer value: 'user' for column
'sh___change_account_id' at row 1

**Comment by benkeen** [comment]
Oops! Better not post passwords here, this is a public place.

Want to email me your database name and I'll log in via that URL you posted and take a look. I'm at:  ben.keen@gmail.com

**Comment by sklmteacher** [comment]
Thanks for your suggestion,
I could prevent errors by changing the values for the database columns as 
sh___change_account_type / enum('admin','client','user','unknown')
sh___change_account_id / mediumtext NULL
However, History of the particular updated submission is not being retrieved in admin login. 
History of the submission made retrievable if the values 'user' in sh___change_account_id changed manually to '0'.

**Comment by crazamahdi** [comment]
I installed submission history module but it is not working for version 3.0.9 any idea how to run that module for latest version. @benkeen

**Comment by crazamahdi** [comment]
On clicking load history(submission history) nothing is displayed it is not working @benkeen
